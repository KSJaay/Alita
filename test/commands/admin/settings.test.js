const {Client} = require("discord.js");
const {successEmbed} = require("../../../utils/embeds");
const settingsCommand = require("../../../commands/admin/settings");
const logger = require("../../../logger");

jest.mock("discord.js");
jest.mock("../../../logger");

const messageSettings = (setting = {}) => {
  if (!setting.enabled) return "Enabled: ❌";

  const {channel, message} = setting;

  return `Enabled: ✅\nChannel: <#${channel}>\nMessage: ${message}`;
};

describe("settings command", () => {
  const client = new Client();

  let interaction;
  let data;

  beforeEach(() => {
    interaction = {
      options: {
        getSubcommand: jest.fn(),
        getChannel: jest.fn(),
        getString: jest.fn(),
      },
      guildId: "123456789",
      reply: jest.fn(),
      guild: {
        channels: {
          cache: {
            get: jest.fn().mockReturnValue({send: jest.fn()}),
          },
        },
      },
      user: {
        id: "987654321",
        username: "test",
      },
    };
    data = {
      guild: {
        goodbye: {
          channel: "987654321",
          message: "Goodbye, {user.name}!",
        },
        welcome: {
          channel: "123456789",
          message: "Welcome, {user.name}!",
        },
      },
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should reply with a success message", async () => {
    await settingsCommand.execute(client, interaction, data);

    expect(interaction.reply).toHaveBeenCalledWith({
      embeds: [
        successEmbed({
          description: `Use admins commands to change these settings`,
          fields: [
            {
              name: `Welcome settings`,
              value: messageSettings(data.guild.welcome),
              inline: true,
            },
            {
              name: `Goodbye settings`,
              value: messageSettings(data.guild.goodbye),
              inline: true,
            },
          ],
        }),
      ],
    });
  });

  it("should call logger.error if an error is thrown", async () => {
    interaction.reply.mockImplementation(() => {
      throw new Error("test");
    });

    await settingsCommand.execute(client, interaction, data);

    expect(logger.error).toHaveBeenCalledWith(
      `Error executing 'settings' command!`,
      {
        label: "Command",
        message: "test",
        data,
      }
    );
  });

  it("snaphot of interaction data", () => {
    expect(settingsCommand.interaction).toMatchSnapshot();
  });
});
