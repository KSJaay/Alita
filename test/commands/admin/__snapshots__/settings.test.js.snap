// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings command snaphot of interaction data 1`] = `
{
  "description": "Information about the current settings",
  "name": "settings",
  "options": [],
}
`;
